src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1434,7 +1434,12 @@ void notifyListeners(List<FlowNode> nodes, boolean synchronous) {
                 for (FlowNode node : nodes) {
                     for (GraphListener listener : toRun) {
                         if (listener instanceof GraphListener.Synchronous == synchronous) {
-                            listener.onNewHead(node);
+                            try {
+                                listener.onNewHead(node);
+                            } catch ( Exception e ) {
+                                LOGGER.fine( "skip error with listener " + listener.getClass().getName()
```

Comment:
You really want to log at a WARNING level for this -- if the listener is failing it usually indicates something fatal going wrong with the Pipeline.

Reply:
yup I agree and done.